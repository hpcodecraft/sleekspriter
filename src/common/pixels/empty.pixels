{"size":[40,30],"frames":[4,4],"layers":[[16,16,"layer 16",0,100,1],[15,15,"layer 15",0,100,1],[14,14,"layer 14",0,100,1],[13,13,"layer 13",0,100,1],[12,12,"layer 12",0,100,1],[11,11,"layer 11",0,100,1],[10,10,"layer 10",0,100,1],[9,9,"layer 9",0,100,1],[8,8,"layer 8",0,100,1],[7,7,"layer 7",0,100,1],[6,6,"layer 6",0,100,1],[5,5,"layer 5",0,100,1],[4,4,"layer 4",0,100,1],[3,3,"layer 3",0,100,1],[2,2,"layer 2",0,100,1],[1,1,"layer 1",0,100,1]],"animations":[],"pixels":[]}