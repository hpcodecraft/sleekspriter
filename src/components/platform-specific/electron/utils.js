import { writeFile } from "fs";
import { basename, dirname } from "path";

import config from "../../../config";
import { stateToFile } from "../../../utils";

export const save = (path, contents, errorHandler) => {
  writeFile(path, JSON.stringify(stateToFile(contents)), function(error) {
    if (error) {
      errorHandler(error);
    }
  });
};

export const fileInfoFromPath = path => {
  const fileName = basename(path, "." + config.fileExtension); // the file name without extension, e.g. "coin"
  const folder = dirname(path); // the complete absolute folder (same as path without the file name)

  return {
    fileName,
    folder,
  };
};

export const fileDefaultPath = (file, extension) => {
  const folder = file.folder || config.defaultFolder;
  const name = file.name || config.defaultName;
  const ext = extension || config.fileExtension;

  return `${folder}/${name}.${ext}`; // TODO: use platform-independent folder seperator
};
