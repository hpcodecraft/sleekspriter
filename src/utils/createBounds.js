import { selectionIsActive } from "./";

const createBounds = (size, selection) => {
  if (!selection || !selectionIsActive(selection)) {
    return {
      start: { x: 1, y: 1 },
      end: { x: size.width, y: size.height },
    };
  } else {
    return selection;
  }
};

export default createBounds;
