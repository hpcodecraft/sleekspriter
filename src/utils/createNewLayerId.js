import { maxBy } from "lodash";

const createNewLayerId = layers => {
  return maxBy(layers, "id").id + 1;
};

export default createNewLayerId;
