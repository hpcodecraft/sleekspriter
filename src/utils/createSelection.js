// this function takes two points and returns a selection object where the
// start always is the top left corner and
// end is the bottom right one
// this makes drawing the selection easier and ensures width calculations etc are always consistent

const createSelection = (p1, p2) => {
  return {
    start: {
      x: p1.x < p2.x ? p1.x : p2.x,
      y: p1.y < p2.y ? p1.y : p2.y,
    },
    end: {
      x: p1.x > p2.x ? p1.x : p2.x,
      y: p1.y > p2.y ? p1.y : p2.y,
    },
  };
};

export default createSelection;
