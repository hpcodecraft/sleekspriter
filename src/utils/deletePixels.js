const deletePixels = (allPixels, frame, layer, pixelsToDelete) => {

  Object.keys(pixelsToDelete).map(x => {
    Object.keys(pixelsToDelete[x]).map(y => {
      // console.log("delete! ", frame, layer, x, y);
      delete allPixels[frame][layer][x][y];
    });

    if(Object.keys(allPixels[frame][layer][x]).length === 0) {
      // console.log("delete! ", frame, layer, x);
      delete allPixels[frame][layer][x];
    }
  });

  if(Object.keys(allPixels[frame][layer]).length === 0) {
    // console.log("delete! ", frame, layer);
    delete allPixels[frame][layer];
  }
  if(Object.keys(allPixels[frame]).length === 0) {
    // console.log("delete! ", frame);
    delete allPixels[frame];
  }

  return allPixels;
};

export default deletePixels;
