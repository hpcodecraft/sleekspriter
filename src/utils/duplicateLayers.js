const duplicateLayers = (layers, targetFrame, nextLayerId) => {
  const newLayers = [];
  for (let i = 0; i < layers.length; i++) {
    const newId = nextLayerId + i;
    const newLayer = Object.assign({}, layers[i]);
    newLayer.id = newId;
    newLayer.frame = targetFrame;
    newLayer.name = `${newLayer.name} copy`;
    newLayers.push(newLayer);
  }

  return newLayers;
};

export default duplicateLayers;
