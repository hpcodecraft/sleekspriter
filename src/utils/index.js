export capitalizeFirstLetter from "./capitalizeFirstLetter";
export createBounds from "./createBounds";
export createNewLayerId from "./createNewLayerId";
export deletePixels from "./deletePixels";
export duplicateLayers from "./duplicateLayers";
export flattenPixels from "./flattenPixels";
export getPivot from "./getPivot";
export getPixelsInFrame from "./getPixelsInFrame";
export getPixelsInScope from "./getPixelsInScope";
export fileToState from "./fileToState";
export insideBounds from "./insideBounds";
export mergeLayerPixels from "./mergeLayerPixels";
export selectionIsActive from "./selectionIsActive";
export stateToFile from "./stateToFile";
export t from "./translate";
