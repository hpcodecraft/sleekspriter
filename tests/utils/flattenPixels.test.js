import pixels from "../__mocks__/layerPixelMock";
import { flattenPixels } from "utils";

jest.unmock("utils");

describe("flattenPixels", () => {
  it("creates an array of pixels from a layer JSON map", () => {
    expect(flattenPixels(pixels)).toMatchSnapshot();
  });
});
