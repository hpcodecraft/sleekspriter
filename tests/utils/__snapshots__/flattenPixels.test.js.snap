// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flattenPixels creates an array of pixels from a layer JSON map 1`] = `
Array [
  Pixel {
    "a": 1,
    "b": 0,
    "frame": 1,
    "g": 0,
    "layer": 1,
    "r": 0,
    "x": 1,
    "y": 1,
  },
  Pixel {
    "a": 1,
    "b": 0,
    "frame": 1,
    "g": 0,
    "layer": 1,
    "r": 0,
    "x": 1,
    "y": 2,
  },
  Pixel {
    "a": 1,
    "b": 0,
    "frame": 1,
    "g": 0,
    "layer": 1,
    "r": 0,
    "x": 2,
    "y": 1,
  },
  Pixel {
    "a": 1,
    "b": 0,
    "frame": 1,
    "g": 0,
    "layer": 1,
    "r": 0,
    "x": 2,
    "y": 2,
  },
]
`;
